README.md

```diff
@@ -1,11 +1,432 @@
+supported on Ruby versions 3.3 and newer.
+
+### Implementing an Activity
+
+Implementing workflows is not yet supported in the Ruby SDK, but implementing activities is.
+
+For example, if you have a `SayHelloWorkflow` workflow in another Temporal language that invokes `SayHello` activity on
+`my-task-queue` in Ruby, you can have the following Ruby script:
+
+```ruby
+require 'temporalio/activity'
+require 'temporalio/cancellation'
+require 'temporalio/client'
+require 'temporalio/worker'
+
+# Implementation of a simple activity
+class SayHelloActivity < Temporalio::Activity
+  def execute(name)
+    "Hello, #{name}!"
+  end
+end
+
+# Create a client
+client = Temporalio::Client.connect('localhost:7233', 'my-namespace')
+
+# Create a worker with the client and activities
+worker = Temporalio::Worker.new(
+  client:,
+  task_queue: 'my-task-queue',
+  # There are various forms an activity can take, see specific section for details.
+  activities: [SayHelloActivity]
+)
+
+# Run the worker until SIGINT. This can be done in many ways, see specific
+# section for details.
+cancellation, cancel_proc = Temporalio::Cancellation.new
+Signal.trap('INT') { cancel_proc }
+worker.run(cancellation:)
+```
+
+Running that will run the worker until Ctrl+C pressed.
+
+### Running a Workflow
+
+Assuming that `SayHelloWorkflow` just calls this activity, it can be run like so:
+
+```ruby
+require 'temporalio/client'
+
+# Create a client
+client = Temporalio::Client.connect('localhost:7233', 'my-namespace')
+
+# Run workflow
+result = client.execute_workflow(
+  'SayHelloWorkflow',
+  'Temporal',
+  id: 'my-workflow-id',
+  task_queue: 'my-task-queue'
+)
+puts "Result: #{result}"
+```
+
+This will output:
+
+```
+Result: Hello, Temporal!
+```
+
+## Usage
+
+### Client
+
+A client can be created and used to start a workflow or otherwise interact with Temporal. For example:
+
+```ruby
+require 'temporalio/client'
+
+# Create a client
+client = Temporalio::Client.connect('localhost:7233', 'my-namespace')
+
+# Start a workflow
+handle = client.start_workflow(
+  'SayHelloWorkflow',
+  'Temporal',
+  id: 'my-workflow-id',
+  task_queue: 'my-task-queue'
+)
+
+# Wait for result
+result = handle.result
+puts "Result: #{result}"
+```
+
+Notes about the above code:
+
+* Temporal clients are not explicitly closed.
+* To enable TLS, the `tls` option can be set to `true` or a `Temporalio::Client::Connection::TLSOptions` instance.
+* Instead of `start_workflow` + `result` above, `execute_workflow` shortcut can be used if the handle is not needed.
+* The `handle` above is a `Temporalio::Client::WorkflowHandle` which has several other operations that can be performed
+  on a workflow. To get a handle to an existing workflow, use `workflow_handle` on the client.
+* Clients are thread safe and are fiber-compatible (but fiber compatibility only supported for Ruby 3.3+ at this time).
+
+#### Cloud Client Using mTLS
```

Comment:
Should also document API Key

Reply:
API key is not GA yet and we don't document it in Python or .NET SDK READMEs, nor do we document in TypeScript places like https://docs.temporal.io/develop/typescript/temporal-clients#connect-to-temporal-cloud. I would expect API key documentation for each language to be done across languages when available/ready.

Reply:
Not GA, but it's Public Preview, which mandates that we should have basic documentation. And I expect it will be a few months before we add Ruby docs on docs.temporal.io. Till then, this README file is the main doc.

But I honestly don't mind. Your call.

Reply:
:+1: We should update all languages together, Ruby should not be special here.